src/PAModelTests/DataSourceTests.cs

```diff
@@ -117,6 +117,25 @@ public void TestConnectionInstanceIDHandling(string appName)
             errorsCaptured.ThrowOnErrors();            
         }
 
+        [DataTestMethod]
+        [DataRow("NameOne")]
+        [DataRow("NameTwo>")]
+        [DataRow("NameThree><*/")]
+        public void TestGetExtensionEncoded(string fileString)
+        {
+            // Make sure exception is not thrown
+            try
+            {
+                FilePath filePath = new FilePath(fileString);
+                filePath.GetExtension();
+            }
+            // If it is thrown, Fail test
+            catch (System.ArgumentException)
+            {
+                Assert.Fail();
```

Comment:
I don't think this will do what you think it will. `Path.GetExtension()` only throws an `ArgumentException` on .NET Framework and .NET Core older than 2.1 ([doc reference](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getextension?view=netcore-3.1)). The unit test project is on .NET Core 3.1, so even if it wasn't encoding the characters it wouldn't throw an exception.

I think you'll need to compare the result from `filePath.GetExtension()` with the espected escaped result.